lib/howl/commands/file_commands.moon

```diff
@@ -70,14 +78,19 @@ command.alias 'save', 'w'
 command.register
   name: 'save-as',
   description: 'Saves the current buffer to a given file'
-  input: 'file'
-  handler: (file) ->
+  interactive: true
+  evade_history: true
+  handler: ->
+    file = interact.select_file allow_new: true
+    return unless file
     if file.exists
-      input = inputs.yes_or_no false
-      unless inputs.read input, prompt: "File '#{file}' already exists, overwrite? "
+      unless interact.yes_or_no prompt: "File '#{file}' already exists, overwrite? "
         log.info "Not overwriting; buffer not saved"
         return
 
+    unless file.parent.exists
```

Comment:
This shows the new path with [New] in the list, giving a good indication of what would happen. I'd suggest you try it out and if you still think it's necessary, I'll inject a confirmation question.

Agree about consistency for `save`. That probably should ask for confirmation for new paths.